src/dask_awkward/layers/layers.py

```diff
@@ -213,11 +187,11 @@ def __init__(
         self.fn = fn
         super().__init__(mapping, **kwargs)
 
-    def mock(self) -> tuple[MaterializedLayer, Any | None]:
+    def mock(self) -> MaterializedLayer:
         mapping = copy.copy(self.mapping)
         if not mapping:
             # no partitions at all
-            return self, None
```

Comment:
Here's a line where I'm wondering if we should return `self, {}` to keep the `mock` interface something that returns a two element tuple

Reply:
I've taken a slightly different tack in e0827e9 — add a new `prepare_for_projection` method that `mock` can call.